### PR TITLE
Python 3 bugfix in ccpp-framework, ESMF 8.1.0bs27, cleanup rt_utils.sh, fix RRTMGP regression tests

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR:
- contains a submodule pointer update for fv3atm for a Python 3 bugfix in ccpp-framework
- uses the correct Python interpreter to run `ccpp_prebuild.py` from `CMakeLists.txt`
- removes machine dependent logic from `tests/rt_utils.sh`
- updates ESMF from 8.1.0bs21 to 8.1.0bs27 on all tier-1 platforms, and all tier-2 systems that I am responsible for
- fixes the RRTGMP regression tests (these tests were running the original RRTMG schemes due to typos in the regression test files); however, since these currently crash (@dustinswales is working on it), they are temporarily turned off in `tests/rt.conf` and `tests/rt_gnu.conf`
- minor updates of pull request template
- turn on INLINE_POST for hera.gnu, cheyenne.intel and cheyenne.gnu in gnumake build (they are already on in the cmake build)

The update from ESMF 8.1.0bs21 to ESMF 8.1.0bs27 (unexpectedly) changes the results of the regression tests. No changes in the actual results, but in the metadata of the `phyf*.{nc,nemsio}` and `dynf*.{nc,nemsio}` files:
- order of global and variable attributes
- the `time` variable (counting hourse since forecast initialization) sometimes - but not always, seems to depend on the regridding option - changes from `double` to `float`

## Testing

- full regression tests passed for new baseline (20200914) on hera.intel, hera.gnu, cheyenne.intel, cheyenne.gnu, orion.intel, wcoss_cray, wcoss_dell_p3
- code compiles successfully on jet.intel when `compile_cmake.sh` is used; however, when using `rt.sh`, there is a problem with line 215 in `FV3/ccpp/physics/CMakeLists.txt`; investigation ongoing, will be fixed in a forthcoming commit
- code compiles and runs on macOS with clang+gfortran and gcc+gfortran
**STILL TODO/CONFIRM**
- manual tests will be run on gaea.intel, stampede.intel
- tests should also be run on the new wcoss platform (which shares the PBS scheduler logic with cheyenne in `tests/rt_utils.sh`)

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/171
- waiting on https://github.com/NCAR/ccpp-framework/pull/322

### CODE REVIEW

File: tests/rt_rrtmgp.conf
Comment (DusanJovic-NOAA) [inline tests/rt_rrtmgp.conf]: Is this file just for debugging rrtmgp tests?

**Comment by climbfuji** [comment]
Note: temporarily added `tests/rt_rrtmgp.conf` (by mistake), will remove before the PR is merged.

File: tests/rt_rrtmgp.conf
Comment (climbfuji) [reply tests/rt_rrtmgp.conf]: See https://github.com/ufs-community/ufs-weather-model/pull/201#issuecomment-691176613 - yes.

File: .github/pull_request_template.md
Comment (MinsukJi-NOAA) [inline .github/pull_request_template.md]: Utests can be run on Hera, Orion, Dell with Intel compilers (I had some trouble with ecFlow on Cray)

Comment (climbfuji) [reply .github/pull_request_template.md]: Should I add this information to the PR template?

Comment (MinsukJi-NOAA) [reply .github/pull_request_template.md]: Since regression tests are supported on all platforms (tier 1, 2, etc.), maybe it would be helpful to add that utests are only supported on tier 1 (Cray will have to come later)? e.g. unit tests (utests on tier 1 platform)

**Comment by climbfuji** [comment]
@MinsukJi-NOAA thanks for the docker/ci updates, the automated tests all passed

**Review by MinsukJi-NOAA — APPROVED** [review]

File: conf/configure.fv3.cheyenne.gnu
Comment (junwang-noaa) [inline conf/configure.fv3.cheyenne.gnu:92]: So the inline post will not be supported on Cheyenne and hera.gnu any more?

File: tests/rt.conf
Comment (junwang-noaa) [inline tests/rt.conf:122]: Will you add back the rrtmgp tests and remove rt_rrtmgp.conf?

**Review by junwang-noaa — APPROVED** [review]

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA this PR is ready from my side. Waiting for CI tests to complete.

**Pull request merged**
